improver/calibration/samos_calibration.py

```diff
@@ -404,11 +406,20 @@ def calculate_statistic_by_rolling_window(self, input_cube: Cube):
                 padded_cubelist.append(new_slice)
         padded_cube = padded_cubelist.concatenate_cube()
 
+        # A data sufficiency check is performed to ensure that each rolling window
+        # contains enough valid data points to produce a sensible result. Data is
+        # considered sufficient if at least 50% of the data points in the window are
+        # valid. Where data is considered insufficient, the result of the rolling
+        # window calculation is replaced with nan.
+        check_data = padded_cube.copy()
+        check_data.data = ~isnan(check_data.data) * 1.0
+        valid_count = self.apply_aggregator(check_data, SUM)
+
         aggregated_cubes = {}
         for aggregator in [MEAN, STD_DEV]:
-            aggregated_cubes[aggregator.name()] = self.apply_aggregator(
-                padded_cube, aggregator
-            )
+            aggregated_cube = self.apply_aggregator(padded_cube, aggregator)
+            aggregated_cube.data[valid_count.data <= pad_width] = nan
```

Comment:
I think I'd prefer to define a separate variable to use here for defining the number of counts required for a rolling window period to be considered valid. You could just define a variable in the `__init__` without necessarily providing support to pass it in from elsewhere right now, as the appetite for changing this is unclear. 

Therefore you could define a variable in the `__init__` e.g. `self.valid_rolling_window_fraction = 0.5` and then define e.g. `allowed_valid_count = int((self.window_length-1) * self.valid_rolling_window_fraction)` to use when setting rolling window periods that don't meet the required number of valid counts to NaN. I think re-using `pad_width` makes things a bit confusing.

Reply:
Added `valid_rolling_window_fraction` input to `__init__` and unit tests to check that the provided value for this is valid. I've not made this available in the CLI as, like you, I'm not sure that there is a need to modify this value at the moment.